importlab/resolve.py

```diff
@@ -95,10 +102,53 @@ def convert_to_path(name):
     return (filename, dot_count)
 
 
+def infer_module_name(filename, fspath):
```

Comment:
Would this make sense as a method on FileSystem? Then instead of checking for the existence of root, you could put the default implementation on FileSystem and the real implementation on OSFileSystem.

Reply:
good idea. i'll give that a try.

Reply:
moved the relative name calculation to FileSystem